org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java

```diff
@@ -145,7 +161,7 @@ public void onBluetoothStateChangedEvent(BluetoothStateChangedEvent event) {
     @OnClick(R.id.activity_obd_selection_layout_rescan_bluetooth)
     protected void rediscover() {
         mBluetoothHandler.stopBluetoothDeviceDiscovery();
-        updateContentView();
+        requestLocationPermissions();
```

Comment:
As @asaikarthikeya explained in the comment above, the GPS dialog is shown twice if adapting your code changes. 
>If we try to turn off the bluetooth not using app and then on activity created it will still show the paired device so we need to update the content

@arvindnegi1: I'm sorry, I could not observe your above described behaviour. Is this still a thing?

Reply:
@SebaDro Sir, I accidentally removed updateContentView() from onBluetoothStateChangedEvent(). Now its working fine.